projects/packages/scheduled-updates/src/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-update-schedules.php

```diff
@@ -328,6 +328,14 @@ public function delete_item( $request ) {
 	 * @return bool|WP_Error
 	 */
 	public function validate_plugins_param( $plugins ) {
+		if ( count( $plugins ) > 10 ) {
+			return new WP_Error(
+				'rest_invalid_param',
+				__( 'You can schedule a maximum of 10 plugins at one time.', 'jetpack-scheduled-updates' ),
+				array( 'status' => 400 )
+			);
+		}
```

Comment:
The REST API supports a `maxItems` argument in the endpoint's args definition, which we should be able to use here?
https://github.com/WordPress/WordPress/blob/44c2d5685a161b9a7b52a8f743e7fb96b2cf5666/wp-includes/rest-api.php#L2464-L2478

Reply:
Hmmm for some reason it's not working after setting `maxItems`. I don't see any similar usage in `jetpack` repo either, only on wpcom codebase. Will take a closer look at this tomorrow.

Reply:
It looks like when a validation callback is provided the built-in schema validation gets skipped. https://github.com/Automattic/jetpack/pull/36375/commits/89f01c0b92f420abd97123d4823247e0dc9d948a brings that back "manually". 

What's odd is that I tried to pass a 'pattern' parameter in the `items` array, [like the wp/v2/plugins endpoint does](https://github.com/WordPress/WordPress/blob/master/wp-includes/rest-api/endpoints/class-wp-rest-plugins-controller.php#L94), but I was not able to make it fail. Maybe I'm missing something.

Reply:
> It looks like when a validation callback is provided the built-in schema validation gets skipped. https://github.com/Automattic/jetpack/commit/89f01c0b92f420abd97123d4823247e0dc9d948a brings that back "manually".

Thanks for jumping in and making the changes!

> What's odd is that I tried to pass a 'pattern' parameter in the items array, [like the wp/v2/plugins endpoint does](https://github.com/WordPress/WordPress/blob/master/wp-includes/rest-api/endpoints/class-wp-rest-plugins-controller.php#L94), but I was not able to make it fail. Maybe I'm missing something.

I was doing some testing and noticed something interesting. When I put the `pattern` parameter inside the `items` array, it actually works. But, it looks like the regex in `self::PATTERN` might be the culprit behind our little issue. Just out of curiosity, what examples did you try to catch a fail? I gave `.plugin` and `plugin..file` a shot, and surprisingly, both sneaked past our regex check. It does fail when I try using `./`.

Reply:
I used the same pattern as the plugins endpoint, but maybe that's not needed at all.